### ISSUE TITLE
Direct Property Accessors giving default value

### ISSUE THREAD

**Issue by softwarejoint** [realm/realm-java#6021]
Realm Version: 5.3.0
Android Studio 3.1.3
Build #AI-173.4819257, built on June 4, 2018
JRE: 1.8.0_152-release-1024-b01 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.13.5

Assuming I have a Realm Model

```
public class Contact extends RealmObject {
     public boolean isSaved;
     ...
}

```
lets say I have a realm object

Contact cnObj;  
where (cnObj.isSaved == true)

Initial compilation and run works perfectly. On the 2nd run if I try to access the values

cnObj.isSaved returns default value of boolean 'false'

I am observing this all property types. I have tried rolling back realm/android studio version as of now. 
I need to be sure this is just a android studio compilation bug & not in realm so any release build does not suffer.
Do you have any suggestions?

**Comment by cmelchior** [comment]
Can you try to show the code that returns the wrong result?

**Comment by cmelchior** [comment]
So to be clear:

Everything works fine if you do a full build, but not on subsequent builds?
We recently enabled incremental builds in the Realm transformer and what you are describing sounds like it might do something wrong. You can see the caveats here: https://realm.io/docs/java/latest/#limitations-incremental-builds

That should be easy to test though. If you try to revert back to Realm Java 5.1 and try again

**Comment by softwarejoint** [comment]
Sure, I will check do a couple of builds and post my update. 

However the realm doc says realm build is needed in case of any changes to realm objects. However any change even in xml file triggers those conditions for me.

**Comment by cmelchior** [comment]
Are you able to share your project with us? You can send it privately to help@realm.io

**Comment by softwarejoint** [comment]
yes, it does solve it. however any changes in xml or any java file does bring the issue back. As to the project I would not be able to share it.

**Comment by softwarejoint** [comment]
I am observing this issue in multiple projects. Is there any branch which will have updates to other stuff ignoring the incremental build feature. It is a pain to clean the project each time.

**Comment by shawhkhi** [comment]
اعتذر عن التأخير وشكرا لكم
saadzmee33

في الجمعة، ١٥ يونيو، ٢٠١٨ ٠٨:٥٠ SoftwareJoint - Erlang, XMPP, Ejabberd &
Servers <notifications@github.com> كتب:

**Comment by cmelchior** [comment]
You can try to use the setup described here: https://realm.io/docs/java/latest/#how-do-i-customize-dependecies-defined-by-the-realm-gradle-plugin

Then you can choose a different version for the Realm transformer than the rest, e.g. 5.1, but if it really is a problem in the transformer it would be very nice if we can find out what is going wrong in your project. Is it possible for you to post your project structure + the gradle build files?

**Comment by cmelchior** [comment]
I assume the above answered your question. In the meantime I'll fold this into #6041 as it seems there are some corner cases where the IncrementalTransform doesn't always do the right thing.

**Comment by nhachicha** [comment]
should be fixed in #6058

**Comment by softwarejoint** [comment]
Thank you, will give it a try
